lib/buffer.js

```diff
@@ -832,8 +832,9 @@ Buffer.prototype.writeUIntLE = function(value, offset, byteLength, noAssert) {
   value = +value;
   offset = offset >>> 0;
   byteLength = byteLength >>> 0;
+  var maxBytes = Math.pow(2, 8 * byteLength) - 1;
   if (!noAssert)
-    checkInt(this, value, offset, byteLength, Math.pow(2, 8 * byteLength), 0);
+    checkInt(this, value, offset, byteLength, maxBytes, 0);
```

Comment:
I think Crankshaft and TurboFan are smart enough to move the Math.pow() call to the use site but the baseline compiler probably isn't, which would mean an extra Math.pow() call in the (unoptimized) common case.